File: Go/DP1/2565.go
package main

import (
	"fmt"
	"sort"
)

const LMIT = 1001

var dp [LMIT]int
var in_list [LMIT]int

func max(a int, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	n := 0
	fmt.Scan(&n)
	a_to_b := make([][]int, 0)
	for i := 1; i <= n; i++ {
		a := 0
		b := 0
		temp_slice := make([]int, 0)
		fmt.Scan(&a, &b)
		temp_slice = append(temp_slice, a, b)
		a_to_b = append(a_to_b, temp_slice)
	}

	sort.Slice(a_to_b, func(i, j int) bool {
		if (a_to_b[i][0]) < a_to_b[j][0] {
			return true
		}
		return false
	})

	for i := 0; i < n; i++ {
		dp[i] = 1
	}
	max_line := 0

	for i := 0; i < n; i++ {
		for j := 0; j < i; j++ {
			if a_to_b[i][1] > a_to_b[j][1] {
				dp[i] = max(dp[i], dp[j]+1)
			}
		}
		max_line = max(max_line, dp[i])
	}

	fmt.Println(n - max_line)
}

File: python/DP1/2565.py
import sys
sys.setrecursionlimit(2500)
LMIT = 501
# dp = [ 1 for _ in range(LMIT)]
in_list =[]


a = []
b = []
if __name__ == "__main__":
    n = int(input())
    dp = [ 1 for _ in range(n)]    #자기 자신은 일단 겹치지 않은 전기 줄이라고 생각
    for _ in range(n):   
        in_list.append(list(map(int,input().split())))

    
    in_list.sort()
 
    a_to_b=list(map(lambda x:x[1],in_list))
    
    for i in range(n):
        for j in range(i):
            if a_to_b[i]> a_to_b[j]: #i가 제일 오른쪽에 있다고 생각 하고,j가 작으면 겹치지 않음
                dp[i]=max(dp[i],dp[j]+1) # 해당 i에 겹치지 않은 최대 전기 줄
    
    print(n-max(dp))
            

  





